plugins/module_utils/vardict.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+# (c) 2023, Alexei Znamensky <russoz@gmail.com>
+# Copyright (c) 2023, Ansible Project
+# Simplified BSD License (see LICENSES/BSD-2-Clause.txt or https://opensource.org/licenses/BSD-2-Clause)
+# SPDX-License-Identifier: BSD-2-Clause
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import copy
+
+
+class _Variable(object):
+    NOTHING = object()
+
+    def __init__(self, diff=False, output=True, change=None, fact=False, verbosity=0):
+        self.init = False
+        self.initial_value = None
+        self.value = None
+
+        self.diff = None
+        self._change = None
+        self.output = None
+        self.fact = None
+        self._verbosity = None
+        self.set(output=output, diff=diff, change=change, fact=fact, verbosity=verbosity)
+
+    def getchange(self):
+        return self.diff if self._change is None else self._change
+
+    def setchange(self, value):
+        self._change = value
+
+    def getverbosity(self):
+        return self._verbosity
+
+    def setverbosity(self, v):
+        if not (0 <= v <= 4):
+            raise ValueError("verbosity must be an int in the range 0 to 4")
+        self._verbosity = v
+
+    change = property(getchange, setchange)
+    verbosity = property(getverbosity, setverbosity)
+
+    def set(self, output=None, diff=None, change=None, fact=None, initial_value=NOTHING, verbosity=None):
```

Comment:
Some documentation on what the types should be would be helpful.

Reply:
Done, although users should not be calling any method directly into an instance of `_Variable`, they should be interacting only with the `VarDict` class.

Reply:
True, but if you look at the `VarDict` function that passes everything to this function, you only see `**kwargs` being passed without any details. So if someone looks for documentation, they end up looking here.